packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx

```diff
@@ -42,185 +27,78 @@ export interface CommentMarkdownInputProps {
-        ...comment,
-        children: parent.node.children,
-      };
-      return copy;
-    });
 
-    if (!editCommentId) {
-      updatePostCache(client, postId, { numComments: post.numComments + 1 });
-
-      trackEvent(
-        postAnalyticsEvent(AnalyticsEvent.CommentPost, post, {
-          extra: { commentId: parentCommentId },
-        }),
-      );
-    }
-
-    if (onCommented) {
-      onCommented(comment, !editCommentId, parentCommentId);
-    }
-  };
-
-  const mutation = parentCommentId
-    ? COMMENT_ON_COMMENT_MUTATION
-    : COMMENT_ON_POST_MUTATION;
-  const {
-    mutateAsync: onComment,
-    isLoading: isCommenting,
-    isSuccess,
-  } = useMutation(
-    (variables: SubmitComment) =>
-      requestMethod(graphqlUrl, mutation, variables, {
-        requestKey: JSON.stringify(key),
-      }),
-    {
-      onSuccess: (data) => onSuccess(data?.comment),
-    },
-  );
-
-  useBackgroundRequest(key, {
-    enabled: isCompanion,
-    callback: ({ res }) => onSuccess(res?.comment),
+  const { mutateComment, isLoading, isSuccess } = useMutateComment({
+    post,
+    editCommentId,
+    parentCommentId,
+    onCommented,
   });
 
-  const { mutateAsync: editComment, isLoading: isEditing } = useMutation(
-    (variables: SubmitComment) =>
-      requestMethod(graphqlUrl, EDIT_COMMENT_MUTATION, variables, {
-        requestKey: JSON.stringify(key),
-      }),
-    { onSuccess: (data) => onSuccess(data?.comment) },
-  );
-
-  const onSubmit = (content: string) => {
-    if (editCommentId) {
-      return editComment({ id: editCommentId, content });
-    }
-
-    return onComment({ content, id: parentCommentId ?? postId });
-  };
-
   const onSubmitForm: FormEventHandler<HTMLFormElement> = (e) => {
     e.preventDefault();
 
     const { content } = formToJson<{ content: string }>(e.currentTarget);
 
-    return onSubmit(content);
+    return mutateComment(content);
   };
 
   const onKeyboardSubmit: FormEventHandler<HTMLTextAreaElement> = (e) => {
     const content = e.currentTarget.value;
 
-    return onSubmit(content);
+    return mutateComment(content);
   };
 
   useEffect(() => {
     markdownRef?.current?.textareaRef?.current?.focus();
   }, []);
 
   return (
-    <form action="#" onSubmit={onSubmitForm} className={className?.container}>
+    <form
+      action="#"
+      onSubmit={onSubmitForm}
+      className={className?.container}
+      style={style}
+    >
       <MarkdownInput
         ref={markdownRef}
         className={{
-          tab: classNames('!min-h-[16rem]', className?.tab),
-          input: replyTo && 'mt-0',
+          tab: classNames('!min-h-16', className?.tab),
+          input: classNames(className?.input, replyTo && 'mt-0'),
           profile: replyTo && '!mt-0',
+          container: className?.markdownContainer,
         }}
         postId={postId}
         sourceId={sourceId}
-        showUserAvatar
-        isLoading={isCommenting || isEditing}
+        showUserAvatar={showUserAvatar}
```

Comment:
We don't show user avatar on the input on mobile. Need to be able to control that from outside.